provider/gateway/client.go

```diff
@@ -92,13 +104,20 @@ func (c *client) SubmitManifest(ctx context.Context, host string, mreq *manifest
 	if err != nil {
 		return err
 	}
-	_, _ = io.Copy(ioutil.Discard, resp.Body)
+	responseBuf := &bytes.Buffer{}
+	_, err = io.Copy(responseBuf, resp.Body)
+	if err != nil {
+		return err
```

Comment:
I believe we always want to `resp.Body.Close()` - perhaps do it as a `defer` before the `io.Copy`.  Same goes for later instances of this change in this PR.

Reply:
I have refactored everything to defer